react/components/Modal/README.md

```diff
@@ -32,11 +32,14 @@
 | aria-label              | `string`  | 🚫       |                     | Acessible Modal name. If this name is visible on the screen, prefer to use aria-labelledby                                                                  |
 | aria-labelledby         | `string`  | 🚫       | `vtex-modal__title` | ID of the element that provides the Modal an accessible name. If aria-label and aria-albelledby is not defined, the default here will be the title element. |
 | aria-describedby        | `string`  | 🚫       |                     | ID of the element that provides the Modal an accessible description                                                                                         |
+| initialFocusRef         | `ref`     | 🚫       |                     | Element to be focused first when Modal is open. The default is the first Modal element.                                                                     |
```

Comment:
I think the user can do this without having to tell the modal to do it

Reply:
Yes, it can. But it's not that simple because of the Modal behavior of opening, the user needs to do a [workaround](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node) to check if `ref.currrent` changes since `useEffect(effect, [ref.current])` doesn't work.

This prop is an easy way for the user to control the initial focus, [ReachUI](https://reacttraining.com/reach-ui/dialog/#dialogoverlay-initialfocusref) does this too.

Reply:
You have a test to do this, I got your test and made how the user could do it

```typescript
  describe('initialFocusRef', () => {
    it.only('to focus correctly', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const Component = () => {
        const initialFocusRef = useRef<HTMLDivElement>(null)

        useEffect(() => {
          initialFocusRef.current?.focus()
        }, [])

        return (
          <Modal
            isOpen
            onClose={onClose}
            container={containerModal}
          >
            <Input
              placeholder="Type your name..."
              size="small"
              ref={initialFocusRef}
            />
          </Modal>
        )
      }
      render(<Component />, { container: containerModal })

      const currentFocused = document.activeElement

      expect(currentFocused).toBe(document.querySelector('input'))
    })
  })
```

Reply:
My problem with this prop is that it is a bit strange to tell the modal what you want o focus and another thing is what would happen if I pass the prop and did the `userEffect` that I added to the test for a different element? Who would be focused? Is a bit confusing the usage of it

Reply:
I think it works because the modal is already open :thinking: I did the same thing at the storybook and didn't work

Reply:
@Klynger Fixed and removed the prop. But I had to do a small fix on the Input component to work properly